meson.build

```diff
@@ -14,4 +14,7 @@ libm = cc.find_library('m', required : false)
 
 sources = ['src/hexgl.c', 'src/utils.c', 'src/analysismap.c', 'src/camerachase.c', 'src/shipcontrols.c', 'src/shipeffects.c']
 
-executable('gnome-hexgl', sources, dependencies: [gthree, libm])
+cflags = ['-DTEXTURES_DIR="@0@/textures/"'.format(meson.current_source_dir()),
+          '-DMODELS_DIR="@0@/models/"'.format(meson.current_source_dir())]
```

Comment:
This will not work though will it? I mean, it will work when we run in-tree, but not installed like the flatpak build?
It needs to be a bit smarter to handle both in-tree and installed, no?

Reply:
Yes, I want that part to be discussed. As I see it just works in-tree was not sure if you really want to install those files.

I remember meld doing a series of checks to decide if it was installed or not

https://gitlab.gnome.org/GNOME/meld/blob/master/bin/meld#L74

I think program could have access to DATADIR and check if there are those files, not sure what should be the best approach.

Just give me a short description and I will implement it.

Reply:
I think what we want is to default to reading from DATADIR, but also check for a GNOME_HEXGL_DATADIR env var, then generate a wrapper script for in-tree running which sets the right env var.

Reply:
Done!. Now flatpak runs smoothly and you can always change the path to in-tree with:

GNOME_HEXGL_DATADIR=/your/project/path gnome-hexgl